app/helpers/routing_helper.rb

```diff
@@ -6,54 +6,106 @@ def self.format_seconds_as_minsec(sec)
+  # rubocop:disable Metrics/MethodLength
+  def self.room_building(input, floor, max_indoor_dist)
+    if valid_coordinates?(input) || BuildingMapHelper.location?(input)
+      return room_building_from_coords(input, floor, max_indoor_dist)
+    end
+    if BuildingMapHelper.building?(input) # in this case we only use outside routing
+      return { indoor: false, building: BuildingMapHelper.map_building_name_to_graph(input), door: nil }
+    end
+    return { indoor: false, building: nil, door: nil } unless BuildingMapHelper.room?(input)
+
+    room = BuildingMapHelper.find_room(input)
+    floor = room.floor
+    door = IndoorRoutingHelper.closest_node([room.location_latitude, room.location_longitude],
+                                            IndoorGraph::BUILDINGS, max_indoor_dist, floor)
+    { indoor: true, building: door[:building], door: door[:door] }
+  end
+  # rubocop:enable Metrics/MethodLength
 
-    begin
-      response = HTTParty.get(routing_url(start, destination))
-      return unless response.code == 200 # OPTIMIZE: give User feedback
+  def self.room_building_from_coords(input, floor, max_indoor_dist)
+    coords = coordinates_from_string(resolve_coordinates(input))
+    door = IndoorRoutingHelper.closest_node(coords, IndoorGraph::BUILDINGS, max_indoor_dist, floor)
+    return { indoor: false, building: nil, door: nil } if door.nil?
 
-      JSON.parse(response.body)["routes"][0]
-    rescue StandardError
-      # OPTIMIZE: give User feedback
-    end
+    { indoor: true, building: door[:building], door: door[:door] }
   end
 
-  def self.transform_route_to_time_marker(route)
-    walking_time = format_seconds_as_minsec(route["duration"])
-    start = route["geometry"]["coordinates"][0]
+  def self.routing_marker(latlng, walkingtime)
+    walkingtime = format_seconds_as_minsec(walkingtime)
     {
-      latlng: [start.second, start.first],
+      latlng: latlng,
       divIcon: {
-        html: walking_time,
+        html: walkingtime,
         className: "time-icon"
       }
     }
   end
 
-  def self.transform_route_to_polyline(route)
-    coordinates = route["geometry"]["coordinates"].map do |(long, lat)|
-      [lat, long]
+  def self.coordinates_from_string(latlng_string)
+    [latlng_string.split(',')[0].to_f, latlng_string.split(',')[1].to_f]
+  end
+
+  def self.best_entry(building, latlng)
+    entry_id = nil
+    min_dist = Float::MAX
+    IndoorGraph.entry_nodes[building].each do |id|
+      distance = distance(IndoorGraph.indoor_graphs[building][id]['latlng'], latlng)
+      if distance < min_dist
+        entry_id = id
+        min_dist = distance
+      end
     end
-    { latlngs: coordinates, options: { className: "routing-path" } }
+    { id: entry_id, latlng: IndoorGraph.indoor_graphs[building][entry_id]['latlng'] }
+  end
+
+  def self.distance(latlng1, latlng2)
+    dy = 111.3 * (latlng1[0] - latlng2[0])
+    dx = 71.5 * (latlng1[1] - latlng2[1])
+    Math.sqrt((dx * dx) + (dy * dy)) * 1000
```

Comment:
Why was the conversation resolved? No one could understand what this does (at least refer to the link from the comment)